src/instructlab/configuration.py

```diff
@@ -1142,8 +1142,13 @@ def ensure_storage_directories_exist() -> bool:
 
     fresh_install = recreate_system_profiles()
 
-    # create expert_args file for users to see/edit
-    if not os.path.isfile(DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE):
+    # open file in read mode to load current contents
+    with open(DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE, "r", encoding="utf-8") as infile:
+        current_dict = yaml.load(infile) or {}
+
+    # check if if there are new keys in defaults
+    if not DEFAULTS.ADDITIONAL_ARGS_DEFAULTS.keys() <= current_dict.keys():
+        # open file in write mode and update contents if there are new keys
```

Comment:
why not just always overwrite this file with the defaults?

Reply:
I'd like to minimize writes if the file exists. This gets run alot

Reply:
should the file read be in a try/except or will it properly set `current_dict` to {} if the file doesn't exist?

Reply:
current_dict will be {} if it doesn't exist I believe